include/AudioPortAudio.h

```diff
@@ -25,137 +25,78 @@
 #ifndef LMMS_AUDIO_PORTAUDIO_H
 #define LMMS_AUDIO_PORTAUDIO_H
 
-#include <QObject>
-
+#include "LcdSpinBox.h"
 #include "lmmsconfig.h"
-#include "ComboBoxModel.h"
 
 #ifdef LMMS_HAVE_PORTAUDIO
 
-#   include <portaudio.h>
-
-#   include "AudioDevice.h"
-#   include "AudioDeviceSetupWidget.h"
-
-#   if defined paNeverDropInput || defined paNonInterleaved
-#	    define PORTAUDIO_V19
-#   else
-#	    define PORTAUDIO_V18
-#   endif
+#include <QObject>
+#include <portaudio.h>
 
-#endif
+#include "AudioDevice.h"
+#include "AudioDeviceSetupWidget.h"
 
+class QComboBox;
 
 namespace lmms
 {
 
-class AudioPortAudioSetupUtil : public QObject
+class PortAudioInitializationGuard
 {
-Q_OBJECT
-public slots:
-	void updateBackends();
-	void updateDevices();
-	void updateChannels();
-
 public:
-	ComboBoxModel m_backendModel;
-	ComboBoxModel m_deviceModel;
+	PortAudioInitializationGuard() { Pa_Initialize(); }
+	~PortAudioInitializationGuard() { Pa_Terminate(); }
```

Comment:
Per PortAudio documentation:
> Note that if `Pa_Initialize()` returns an error code, `Pa_Terminate()` should NOT be called.

That's also what `AudioPortAudio::~AudioPortAudio` was doing.

Reply:
Yeah, I probably should check for errors in there.